telemetry/inflight/proxy.go

```diff
@@ -47,7 +47,8 @@ type ProxyTracer struct {
 
 func (t ProxyTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
 	ctx, span := t.tracer.Start(ctx, spanName, opts...)
-	return ctx, proxySpan{sp: span, onUpdate: t.onUpdate}
+	span = proxySpan{sp: span, onUpdate: t.onUpdate}
+	return trace.ContextWithSpan(ctx, span), span
```

Comment:
@vito this was just something I found while in the area... I found I needed this for the `buildkitTracer` I made, and thought it *probably* makes sense here as well?

That said, I'm not 100% sure what kind of effect this would have - my guess is that previously we might have ended up with child spans that wouldn't have appeared as "live" updated spans?